homeassistant/components/airpatrol/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for the AirPatrol integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from airpatrol.api import AirPatrolAPI, AirPatrolAuthenticationError, AirPatrolError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.EMAIL,
+                autocomplete="username",
+            )
+        ),
+        vol.Required(CONF_PASSWORD): TextSelector(
+            TextSelectorConfig(
+                type=TextSelectorType.PASSWORD,
+                autocomplete="current-password",
+            )
+        ),
+    }
+)
+
+
+async def validate_api(
+    hass: HomeAssistant, user_input: dict[str, str]
+) -> tuple[str | None, str | None, dict[str, str]]:
+    """Validate the API connection."""
+    errors: dict[str, str] = {}
+    session = async_get_clientsession(hass)
+    access_token = None
+    unique_id = None
+    try:
+        api = await AirPatrolAPI.authenticate(
+            session, user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+        )
+        access_token = api.get_access_token()
+        unique_id = api.get_unique_id()
+    except AirPatrolAuthenticationError:
+        errors["base"] = "invalid_auth"
+    except AirPatrolError:
+        errors["base"] = "cannot_connect"
+    except Exception:  # noqa: BLE001
+        errors["base"] = "unknown"
+    return (access_token, unique_id, errors)
+
+
+class AirPatrolConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for AirPatrol."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            access_token, unique_id, errors = await validate_api(self.hass, user_input)
+            if access_token and unique_id:
+                # Store the access token in the config entry
+                user_input["access_token"] = access_token
```

Comment:
```suggestion
                user_input[CONF_ACCESS_TOKEN] = access_token
```
`from homeassistant.const import CONF_ACCESS_TOKEN`
Please change everywhere to use the constant